design/external-management.md

```diff
@@ -0,0 +1,120 @@
+# Enable external Ceph management tools
+
+## TL;DR
+
+Some tools want to use Rook to run containers, but not manage the logical
+Ceph resources like Pools.  We should make Rook's pool management optional.
+
+## Background
+
+Currently in Rook 0.8, creating and destroying a Filesystem (or ObjectStore)
+in a Ceph cluster also creates and destroys the associated Ceph filesystem
+and pools.
+
+The current design works well when the Ceph configuration is within the
+scope of what Rook can configure itself, and the user does not modify
+the Ceph configuration of pools out of band.
+
+## Limitations
+
+The current model is problematic in some cases:
+
+- A user wants to use Ceph functionality outside of Rook's subset, and
+  therefore create their pools by hand before asking Rook to run
+  the daemon containers for a filesystem.
+- A user externally modifies the configuration of a pool (such as the
+  number of replicas), they probably want that new configuration, rather than
+  for Rook to change it back to match the Rook Filesystem settings.
+- A risk-averse user wants to ensure that mistaken edits to their Rook config cannot
+  permanently erase Ceph pools (i.e. they want to only delete pools through
+  an imperative interface with confirmation prompts etc).
+
+## Proposal
+
+In FilesystemSpec and ObjectStoreSpec, group the metadata and
+data pool attributes into a new struct called FilesystemPoolsSpec.
```

Comment:
@bassam did you feel strongly about moving the pools under an overall pool spec as you mentioned [here](https://github.com/rook/rook/pull/1900#discussion_r210705969)? I'd like the avoid the migration cost since it seems roughly equivalent to just allowing empty data and metadata pools.

Reply:
@travisn no. I walked away from the [discussion above](
https://github.com/rook/rook/pull/1900#discussion_r211812335) thinking we would just add `omitEmpty` to pool names. No need for `OnlyManageDaemons` or anything else.